web/auth/basic: strictly validate the auth header to prevent garbage characters from
allowing authentication to succeed
--- os-twisted.orig/src/twisted/web/_auth/basic.py
+++ os-twisted/src/twisted/web/_auth/basic.py
@@ -11,7 +11,7 @@ HTTP BASIC authentication.
 """
 
 
-import binascii
+import base64
 
 from zope.interface import implementer
 
@@ -47,8 +47,11 @@ class BasicCredentialFactory:
         L{credentials.UsernamePassword} instance.
         """
         try:
-            creds = binascii.a2b_base64(response + b"===")
-        except binascii.Error:
+            creds = base64.b64decode(
+                response + b"=" * (-len(response) % 4),
+                validate=True,
+            )
+        except ValueError:
             raise error.LoginFailed("Invalid credentials")
 
         creds = creds.split(b":", 1)
